File: problems/leetcode/11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1, max_water = 0;
        while (left < right) {
            max_water = max(max_water, (right - left) * min(height[left], height[right]));
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return max_water;
    }
};

File: problems/leetcode/11.py
class Solution(object):
    def maxArea(self, height):
        max_heighest = 0
        l, r = 0, len(height) - 1
        while l<r:
            width = r-l
            temp_height = min(height[l], height[r])
            value = width*temp_height
            max_heighest = max(max_heighest, value)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return max_heighest